### PR TITLE
FEATURE: save expiring data to redis

### PR DESCRIPTION
**Pull request by gx578007**

### CODE REVIEW

File: pkg/service/persistence.go [hunk 1]
```diff
@@ -10,6 +12,10 @@ type Store interface {
 	Reset() error
 }
 
+type ExpiringData interface {
```
Comment (c9s) [inline pkg/service/persistence.go:15]: Expirable ?
https://en.wiktionary.org/wiki/expirable#:~:text=Adjective,being%20brought%20to%20an%20end.

File: pkg/service/persistence.go [hunk 2]
```diff
@@ -10,6 +12,10 @@ type Store interface {
...
+	Expiration() time.Duration
```
Comment (c9s) [inline pkg/service/persistence.go:16]: use TimeToLive() or TTL()?

Expiration() looks like it will return a specific time point?

**Review by c9s — APPROVED** [review]

File: pkg/service/persistence.go (hunk 2 above)
Comment (gx578007) [reply pkg/service/persistence.go:16]: I just followed the naming of redis client. It uses "expiration" as `time.Duration`.

**Comment by gx578007** [comment]
polygon:0x8A19A182Af3f0090a678204e4E4D048C2C2715De

**Pull request merged**